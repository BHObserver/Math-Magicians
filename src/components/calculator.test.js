import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from './calculator';

test('renders calculator with initial state', () => {
  render(<Calculator />);
});

test('', () => {
  render(<Calculator />);

  // Click some buttons to update the display
  fireEvent.click(screen.getByText('1'));
  fireEvent.click(screen.getByText('+'));
  fireEvent.click(screen.getByText('2'));
  fireEvent.click(screen.getByText('='));
});
