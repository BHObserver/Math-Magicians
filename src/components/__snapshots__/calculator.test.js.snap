// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component snapshot 1`] = `
<div
  className="container"
>
  <div
    className="text-container"
  >
    <h1>
      Let's Do Some Math!
    </h1>
  </div>
  <div
    className="cal-container"
  >
    <span
      className="output"
    >
      0
       
       
    </span>
    <span
      className="btn-container-one"
    >
      <button
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="bg-color"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </span>
    <span
      className="btn-container-two"
    >
      <button
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="bg-color"
        name="x"
        onClick={[Function]}
        type="button"
      >
        ร
      </button>
    </span>
    <span
      className="btn-container-three"
    >
      <button
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="bg-color"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </span>
    <span
      className="btn-container-four"
    >
      <button
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="bg-color"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </span>
    <span
      className="btn-container-five"
    >
      <button
        className="zero"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="bg-color"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </span>
  </div>
</div>
`;
